<a href="https://colab.research.google.com/github/Anthea05/Anthea_machine-learning/blob/main/TG1_244107020024_AntheaAmodia.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
pip install pyecg --no-deps --break-system-packages && pip install ishneholterlib PyCRC nose pytest-cov --break-system-packages

  Using cached pyECG-0.0.9.1-py2.py3-none-any.whl.metadata (3.1 kB)
  Using cached pycrc-0.11.0-py3-none-any.whl.metadata (2.7 kB)
  Using cached nose-1.3.7-py3-none-any.whl.metadata (1.7 kB)
  Using cached pytest_cov-7.1.0-py3-none-any.whl.metadata (32 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 154.7/154.7 kB 13.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 257.2/257.2 kB 19.4 MB/s eta 0:00:00
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pyecg 0.0.9.1 requires wfdb==2.2.1, which is not installed.
pyecg 0.0.9.1 requires ishneholterlib==2017.4.11, but you have ishneholterlib 2020.5.29 which is incompatible.


Jelaskan kegunakan masing-masing library tersebut!
1. pyPREP merupakan library Python yang digunakan khusus untuk pra-pemrosesan (preprocessing) data sinyal otak (EEG). Fungsinya untuk membersihkan data mentah EEG sebelum dianalisis lebih lanjut, misalnya mendeteksi channel yang rusak atau noise, interpolasi channel yang bermasalah, dan referencing ulang sinyal. PyPREP ini adalah port dari pipeline standar bernama PREP.
2. SciPy merupakan Library fundamental untuk komputasi ilmiah, matematika, dan teknis di Python. Dalam konteks pengolahan sinyal, SciPy menyediakan fungsi tingkat lanjut yang tidak ada di Python bawaan, seperti filtering sinyal.
3. wandb (Weights & Biases) merupakan platform atau tools yang banyak digunakan dalam Machine Learning untuk melakukan tracking (pelacakan) eksperimen model kecerdasan buatan. contoh Kegunaannya saat kita membuat bikin model klasifikasi (misalnya klasifikasi aritmia dari sinyal ECG) dan mau memantau progres training tanpa harus print manual ke console terus-terusan.
4. pyECG merupakan library kecil yang fokus ke membaca/menulis format dataset ECG yang umum dipakai di riset, seperti format WFDB (dari PhysioNet) dan ISHNE (format standar rekaman Holter monitor).


1. Berikan contoh tentang tidakan melanggar etika dan hukum tentang penggunaan kecerdasan buatan. Sertakan referensi yang digunakan.
Deepfake voice cloning di Indonesia. kasus Baim Wong. Di Indonesia ada kasus deepfake dan voice cloning yang membawa nama baim wong. Didalam kasus ini AI digunakan bukan sebagaimana mestinya. Dalam kasus ini AI digunakan sebagai alat manipulasi dan penipuan online sehingga merugikan korbannya secara psikologis dan materi. Kasus semacam ini menunjukkan kalau tanpa standar etika dan regulasi yang tegas, AI gampang banget jadi alat kejahatan digital.
referensi :
- https://journal.lpkd.or.id/index.php/Progres/article/download/2747/2887/1557
- https://journal.lpkd.or.id/index.php/Progres/article/download/2747/2887/15576

2. Berikan contoh tentang dampak energi dan lingkungan terhadap pemanfaatan kecerdasan buatan dan bagaimana cara mengatasinya (Anda dapat memberikan pendapat pribadi). Sertakan referensi yang digunakan.
Tingginya kebutuhan energi untuk menjalankan server di pusat data (data center) memicu lonjakan emisi karbon yang signifikan pada perusahaan-perusahaan teknologi besar. Tercatat bahwa emisi Microsoft naik sekitar 40% (2020–2023), emisi Meta naik lebih dari 65% (2020–2022), dan emisi Google meningkat hampir 50% (2019–2023) yang disebabkan oleh intensitas komputasi AI. Selain energi, pusat data AI juga mengonsumsi air dalam jumlah yang sangat besar untuk mendinginkan server yang panas. Di Amerika Serikat, pusat data rata-rata menggunakan sekitar 7.100 liter air untuk setiap megawatt-jam energi. Pusat data Google di AS sendiri menghabiskan sekitar 12,7 miliar liter air bersih pada tahun 2021.

- https://planetdetroit.org/2024/10/ai-energy-carbon-emissions/
- https://theconversationcompower-hungry-ai-is-driving-a-surge-in-tech-giant-carbon-emissions-nobody-knows-what-to-do-about-it-233452